pkgs/by-name/xr/xrsh/package.nix

```
@@ -0,0 +1,63 @@
+{
+  stdenv,
+  fetchFromGitea,
+  makeWrapper,
+  writeShellApplication,
+  python3Packages,
+  openssl,
+  lib,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xrsh";
+  version = "0.143-unstable-2025-02-28";
+
+  src = fetchFromGitea {
+    fetchSubmodules = true;
+    domain = "codeberg.org";
+    owner = "xrsh";
+    repo = "xrsh";
+    rev = "5d6e9f4615e57f01ba0859000d90f4f162a06653";
+    hash = "sha256-9I6c1GkO7XQZ242ZBCxPiPxkZaYYi/UDLoUvS6nE+M0=";
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase =
+    let
+      launcher = writeShellApplication {
+        name = "xrsh";
+        runtimeInputs = [ python3Packages.twisted ];
+        text = ''
+          XRSH_ROOT="''${1}"
+          XRSH_PORT="''${XRSH_PORT-8080}"
+
+          exec twistd -n web --listen "tcp:''${XRSH_PORT}" --path "''${XRSH_ROOT}"
+        '';
+      };
+    in
+    ''
+      runHook preInstall
+
+      mkdir -p $out/{bin,share/xrsh}
+      cp -r -t $out/share/xrsh index.html src xrsh.js xrsh.ico xrsh.svg
+
+      makeWrapper ${lib.getExe launcher} $out/bin/xrsh \
+        --add-flags "${placeholder "out"}/share/xrsh/src"
+
+      runHook postInstall
+    '';
+
```

This shouldn't be a blocker, but as I've detailed in [another comment](https://github.com/NixOS/nixpkgs/pull/375328#discussion_r1975489291), I suggest adding an update script for convenience:

```suggestion

  passthru.updateScript = unstableGitUpdater { };
  
```

---

thanks! done